daemon/health.go

```diff
@@ -248,12 +248,31 @@ func handleProbeResult(d *Daemon, c *container.Container, result *types.Healthch
 // There is never more than one monitor thread running per container at a time.
 func monitor(d *Daemon, c *container.Container, stop chan struct{}, probe probe) {
 	probeInterval := timeoutWithDefault(c.Config.Healthcheck.Interval, defaultProbeInterval)
+	startInterval := timeoutWithDefault(c.Config.Healthcheck.StartInterval, defaultProbeInterval)
+	startPeriod := timeoutWithDefault(c.Config.Healthcheck.StartPeriod, defaultStartPeriod)
 
 	intervalTimer := time.NewTimer(probeInterval)
 	defer intervalTimer.Stop()
 
+	c.Lock()
+	started := c.State.StartedAt
+	var status string
+	if c.State.Health != nil {
+		status = c.State.Health.Health.Status
+	}
```

Comment:
I think it's safe to assume that `c.State.Health != nil`. `handleProbeResult` makes that assumption, at least.

Reply:
You are correct, it is created before calling `monitor`